defs.h

```diff
@@ -85,6 +85,12 @@
 #define PAL16_RED		224
 #define PAL16_GRAY		240
 
+#define SCREEN_WIDTH	640
+#define SCREEN_HEIGHT	480
+#define BUFFER_WIDTH	768

```

Comment:
Could we make buffer size derived from screen dixe?

Reply:
Sure, but I'm not quite sure what formula they used to calculate it. It probably wasn't `640 + 128`. Given that tiles are 32x32, it could have been `SCREEN_WIDTH + TILE_SIZE * 4`. Height would be `SCREEN_HEIGHT + TILE_SIZE * 2 + 16`. But this is just a guess.

Reply:
From the way the buffer is accessed we know `BUFFER_WIDTH` is `64 + SCREEN_WIDTH + 64`. Though the `64` value is still unclear.
`BUFFER_HEIGHT` appears to be an odd `160 + SCREEN_HEIGHT + 16`.

I'm thinking the extra size of the buffer has to do with sprite clipping seeing as the largest sprites in game are 160px tall.

Reply:
P.s. ROW_PITCH should probably also be replaced with BUFFER_WIDTH

Reply:
I didn't think of that. It would be interesting to change the resolution to show the whole buffer to see what goes on. Interestingly is that the demo just used 640x480, and they increased the buffer size in the beta.

Reply:
I'm guessing that they either added it to allow monsters to enter the screen gradually, or to avoid crashes from writing to memory outside of the buffer.

Reply:
> I'm thinking the extra size of the buffer has to do with sprite clipping seeing as the largest sprites in game are 160px tall, the bottom needing less because it's covered byte the interface.

That's an interesting insight! I think you may be onto something :)

> I'm guessing that they either added it to allow monsters to enter the screen gradually, or to avoid crashes from writing to memory outside of the buffer.

I also consider this to be a rather reasonable explanation.

Reply:
Crashing definitely seems to be most likely, given how terrible we know diablo's engine is it's prone to such. (and that explains the redundant `assert(gpBuffer)` every time the back buffer is written to)